File: src/main/java/WordCounter.java
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class WordCounter {

    // Considered thread safe
    public static final Pattern PATTERN = Pattern.compile("[a-zA-Z]+");
    private final List<String> stopwords;

    public WordCounter(List<String> stopwords) {
        this.stopwords = stopwords;
    }

    public long getWordCount(final String line) {
        if (line == null) {
            return 0;
        }

        Matcher matcher = PATTERN.matcher(line);
        return matcher.results().filter(word -> !stopwords.contains(word.group())).count();
    }
}

File: src/main/kotlin/WordCounter.kt
class WordCounter(val filters: List<SequenceFilter>) {

    internal var regex: Regex = Regex("([a-zA-Z]+)")

    fun countWords(input: String ): List<Int> {
        var currentList = regex.findAll(input).map { it.value }
        val result = filters.map {
            currentList = it.filter(currentList)
            currentList.count()
        }
        return result
    }

}